src/CBLCollection_CAPI.cc

```diff
@@ -82,7 +82,9 @@ CBLScope* CBLDatabase_DefaultScope(const CBLDatabase* db) noexcept {
 }
 
 CBLCollection* CBLDatabase_DefaultCollection(const CBLDatabase* db) noexcept {
-    return const_cast<CBLDatabase*>(db)->getDefaultCollection(false);
+    try {
+        return const_cast<CBLDatabase*>(db)->getDefaultCollection(false);
+    } catchAndWarn()
```

Comment:
Missing final return nullptr ?

Reply:
catchAndWarn() will also return nullptr.

Reply:
I don't think this is a guarantee, and just a result of not having a return statement in the method at that point....it's sort of archaic C behavior that is frowned upon.

Reply:
Nevermind, I was looking at Core's definition of catchAndWarn, not CBL-C